### ISSUE TITLE
About inference speed of nnunet.

### ISSUE THREAD

**Issue by ysssgdhr** [MIC-DKFZ/nnUNet#258]
thank you very much!

Another question ....

I have finished traing 5 models of Unet-fullres , then I runed nnUNet_predict command line yesterday.
And It has been 24 hours , It only finished 18 CT images' prediction. One CT scan for more than one hour.

Is it normal to run that slow ? Or is there any way I can take to boost its speed?
Thank you again!

**Comment by FabianIsensee** [comment]
Hi,
it is normal. sliding window takes ages. Disable test time data augmentation to get a 8x speed improvement at a small loss of accuracy.
Best,
Fabian

**Comment by Bamboo429** [comment]
Hi Fabian,

I faced same issue that inference time is slow, can you tell me how can I disable the test time data augmentation? 
Thank you so much.

Best,
Tracy

**Comment by fhrzic** [comment]
Hi Tracy,

I am not Fabian so my answer could be a bit wrong. Now, you need to disable TTA during inference (that is applying mirroring).

Script: predict_from_raw_data is holding all necessary "things" for prediction. There is flag for TTA(mirroring) which can be disabled when invoking predictions by adding flag: --disable_tta. 

If you want to dig a bit further:

- inference/predict_from_raw_data.py find method _internal_predict_sliding_window_return_logits 

- That method calls  prediction = self._internal_maybe_mirror_and_predict(workon)[0].to(results_device) 

- That method calculates the prediction with mirroring (augmentation). If you override that method with just prediction = self.network(x) return prediction. It should do the trick.

Or set  

- inference_allowed_mirroring_axes = checkpoint['inference_allowed_mirroring_axes'] if \
                    'inference_allowed_mirroring_axes' in checkpoint.keys() else None 
to None without checking config file. 

I am did not test any of these solutions, but I hope they help.

Best,
Franko
